include/substrait-mlir/Dialect/Substrait/IR/SubstraitAttrs.td

```diff
@@ -203,9 +215,25 @@ def Substrait_AtomicAttributes {
   ];
 }
 
+/// Attributes of currently supported compound types, listed in order of
+/// substrait specification.
+def Substrait_CompoundAttributes {
```

Comment:
I argued for using the term "compound" but I think I got mislead by my limited English. "compound" is sort of synonym to "container", no? A fixed char doesn't "compound" anything, does it? Or at least a decimal doesn't "compound" anything. I think what we said at one point is that these are *parametrized*, no?